File: android/src/main/java/com/followanalytics/reactnative/sdk/RNFollowAnalyticsInApp.java
package com.followanalytics.reactnative.sdk;

import android.util.Log;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.followanalytics.FollowAnalytics;

public class RNFollowAnalyticsInApp extends ReactContextBaseJavaModule {
    private final ReactApplicationContext reactContext;

    public RNFollowAnalyticsInApp(ReactApplicationContext reactContext) {
        super(reactContext);
        this.reactContext = reactContext;
    }

    @Override
    public String getName() {
        return "RNFollowAnalyticsInApp";
    }

    @ReactMethod
    public void pauseCampaignDisplay() {
        try {
            FollowAnalytics.InApp.pauseCampaignDisplay();
        } catch (Exception e) {
            Log.d("Error pausing campaigns", e.getMessage());
        }
    }

    @ReactMethod
    public void resumeCampaignDisplay() {
        try {
            FollowAnalytics.InApp.resumeCampaignDisplay();
        } catch (Exception e) {
            Log.d("Error resuming campaign", e.getMessage());
        }
    }

    @ReactMethod
    public void getAll(Callback callback) {
        try {
            callback.invoke(null, FollowAnalytics.InApp.getAll());
        } catch (Exception e) {
            Log.d("Unable to get Campaigns", e.getMessage());
        }
    }

    @ReactMethod
    public void get(String campaignId, Callback callback) {
        try {
            callback.invoke(null, FollowAnalytics.InApp.get(campaignId));
        } catch (Exception e) {
            Log.d("Unable to get Campaigns", e.getMessage());
        }
    }

    @ReactMethod
    public void markAsRead(String[] campaignIds, Callback callback) {
        try {
            FollowAnalytics.InApp.markAsRead(campaignIds);
        } catch (Exception e) {
            Log.d("Unable to get Campaigns", e.getMessage());
        }
    }

    @ReactMethod
    public void markAsUnread(String[] campaignIds, Callback callback) {
        try {
            FollowAnalytics.InApp.markAsUnread(campaignIds);
        } catch (Exception e) {
            Log.d("Unable to mark unread", e.getMessage());
        }
    }

    @ReactMethod
    public void delete(String[] campaignIds, Callback callback) {
        try {
            FollowAnalytics.InApp.delete(campaignIds);
        } catch (Exception e) {
            Log.d("Unable to delete", e.getMessage());
        }
    }


}

File: ios/RNFollowAnalyticsInApp.m
#import "RNFollowAnalyticsInApp.h"
#import <React/RCTLog.h>

@implementation RNFollowAnalyticsInApp

- (dispatch_queue_t)methodQueue
{
    return dispatch_get_main_queue();
}
RCT_EXPORT_MODULE()
// InApp
RCT_EXPORT_METHOD(pauseCampaignDisplay)
{
    [FollowAnalytics.inApp pauseCampaignDisplay];
}

RCT_EXPORT_METHOD(resumeCampaignDisplay)
{
    [FollowAnalytics.inApp resumeCampaignDisplay];
}

RCT_EXPORT_METHOD(getAll:(RCTResponseSenderBlock)callback)
{
    callback(@[[NSNull null], [FollowAnalytics.inApp getAll]]);
}

RCT_EXPORT_METHOD(get:(nonnull NSString *)campaignId callback:(RCTResponseSenderBlock)callback)
{
    callback(@[[NSNull null], [FollowAnalytics.inApp get:campaignId]]);
}

RCT_EXPORT_METHOD(markAsRead:(nonnull NSArray *)campaignIds)
{
    [FollowAnalytics.inApp markAsRead:campaignIds];
}

RCT_EXPORT_METHOD(markAsUnread:(nonnull NSArray *)campaignIds)
{
    [FollowAnalytics.inApp markAsUnread:campaignIds];
}

RCT_EXPORT_METHOD(delete:(nonnull NSArray *)campaignIds)
{
    [FollowAnalytics.inApp deleteIdentifiers:campaignIds];
}
@end
  
